src/transform/src/dataflow.rs

```diff
@@ -375,3 +377,58 @@ pub fn optimize_dataflow_monotonic(dataflow: &mut DataflowDesc) -> Result<(), Tr
 
     Ok(())
 }
+
+fn optimize_dataflow_index_imports(
+    dataflow: &mut DataflowDesc,
+    indexes: &dyn IndexOracle,
+) -> Result<(), TransformError> {
+    let mut indexes_by_view = HashMap::new();
+    for sink_desc in dataflow.sink_exports.iter() {
+        indexes_by_view
+            .entry(sink_desc.1.from)
+            .or_insert_with(HashSet::new);
+    }
+    for (_, (index_desc, _)) in dataflow.index_exports.iter() {
+        indexes_by_view
+            .entry(index_desc.on_id)
+            .or_insert_with(HashSet::new);
+    }
+    for build_desc in dataflow.objects_to_build.iter_mut() {
+        build_desc.plan.as_inner_mut().visit_post(&mut |e| {
+            if let MirRelationExpr::Get {
```

Comment:
Could this `if let` and the other in line 404 be a match instead? Or there is some stylistic consideration to make them `if let`?

Reply:
I can match on `MirRelationExpr::ArrangeBy { input, keys } `, but I can't match on `MirRelationExpr::ArrangeBy { input, keys } if let MirRelationExpr::Get {id: mz_expr::Id::Global(id), ..}` because that feature is still experimental.

```   --> src/transform/src/dataflow.rs:407:60
    |
407 |                   MirRelationExpr::ArrangeBy { input, keys } if let MirRelationExpr::Get {
    |  ____________________________________________________________^
408 | |                     id: mz_expr::Id::Global(id),
409 | |                     ..
410 | |                 } = &**input => {
    | |____________________________^
    |
    = note: see issue #51114 <https://github.com/rust-lang/rust/issues/51114> for more information
    = help: you can write `if matches!(<expr>, <pattern>)` instead of `if let <pattern> = <expr>`
```

I also can't match on `MirRelationExpr::ArrangeBy { input: MirRelationExpr::Get{id, ..} , keys }` because `input` is type `Box<MirRelationExpr>`.

Reply:
Done